desktop/ui/src/main/java/org/datacleaner/windows/HadoopConnectionToNamenodeDialog.java

```diff
@@ -0,0 +1,282 @@
+import javax.swing.event.DocumentEvent;
+
+import org.datacleaner.bootstrap.WindowContext;
+import org.datacleaner.configuration.ServerInformation;
+import org.datacleaner.panels.DCBannerPanel;
+import org.datacleaner.panels.DCPanel;
+import org.datacleaner.server.DirectConnectionHadoopClusterInformation;
+import org.datacleaner.user.MutableServerInformationCatalog;
+import org.datacleaner.util.DCDocumentListener;
+import org.datacleaner.util.ErrorUtils;
+import org.datacleaner.util.IconUtils;
+import org.datacleaner.util.ImageManager;
+import org.datacleaner.util.NumberDocument;
+import org.datacleaner.util.StringUtils;
+import org.datacleaner.util.WidgetFactory;
+import org.datacleaner.util.WidgetUtils;
+import org.datacleaner.widgets.Alignment;
+import org.datacleaner.widgets.DCLabel;
+import org.jdesktop.swingx.JXStatusBar;
+import org.jdesktop.swingx.JXTextField;
+
+/**
+ * Dialog for creating and editing a direct connection to Hadoop namenode
+ */
+public class HadoopConnectionToNamenodeDialog extends AbstractDialog {
+
+    private final DirectConnectionHadoopClusterInformation _directConnection;
+    private final JLabel _statusLabel;
+    private final JXTextField _nameTextField;
+    private final JXTextField _hostTextField;
+    private final JXTextField _portTextField;
+    private final JXTextField _descriptionTextField;
+    private final JButton _saveButton;
+    private final JButton _cancelButton;
+    private final MutableServerInformationCatalog _mutableServerInformationCatalog;
+    private ServerInformation _savedServer = null;
+    private static final ImageManager imageManager = ImageManager.get();
+
+    public HadoopConnectionToNamenodeDialog(WindowContext windowContext,
+            DirectConnectionHadoopClusterInformation directConnection,
+            MutableServerInformationCatalog serverinformationCatalog) {
+        super(windowContext);
+
+        _statusLabel = DCLabel.bright("Please specify connection name");
+        _directConnection = directConnection;
+        _nameTextField = WidgetFactory.createTextField("MyConnection");
+        _hostTextField = WidgetFactory.createTextField("localhost");
+        _portTextField = WidgetFactory.createTextField("9000");
+        _portTextField.setDocument(new NumberDocument(false));
+        _descriptionTextField = WidgetFactory.createTextField("description");
+        _mutableServerInformationCatalog = serverinformationCatalog;
+
+        final String saveButtonText = directConnection == null ? "Register connection" : "Save connection";
+        _saveButton = WidgetFactory.createPrimaryButton(saveButtonText, IconUtils.ACTION_SAVE_BRIGHT);
+        _saveButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+
+                URI nameNodeUri;
+                try {
+                    nameNodeUri = new URI("hdfs", null, _hostTextField.getText(), Integer.parseInt(_portTextField
```

Comment:
I'm not sure that hdfs is always a given. Maybe consider adding a scheme field with a default of HDFS? I think we have a (probably non-exhaustive) list of possible Hadoop schemes somewhere.